agoric/adapter.js

```diff
@@ -0,0 +1,98 @@
+const { Requester, Validator } = require('@chainlink/external-adapter')
+const http = require('http')
+
+const oracleAPI = process.env.AG_SOLO_ORACLE_URL
+if (!oracleAPI) {
+  throw Error(`Must supply $AG_SOLO_ORACLE_URL`)
+}
+const oracleUrl = new URL(oracleAPI)
+
+const customParams = {
+  request_id: ['request_id'],
+  result: ['result'],
+  payment: ['payment'],
+}
+
+const Nat = (n) => {
+  if (!Number.isSafeInteger(n)) {
+    throw Error(`${n} is not a safe integer`)
+  }
+  return n
+}
+
+// FIXME: Ideally, these would be the same.
+const LINK_DECIMALS = 18
+const LINK_AGORIC_DECIMALS = 6
+if (LINK_AGORIC_DECIMALS > LINK_DECIMALS) {
+  throw Error(
+    `LINK_AGORIC_DECIMALS ${LINK_AGORIC_DECIMALS} must be less than or equal to ${LINK_DECIMALS}`,
+  )
+}
+
+const getRequiredFee = (value) => {
+  const str = String(value || 0)
+  const digits = str
+  const significant = digits.substr(
+    0,
+    Math.max(0, digits.length - (LINK_DECIMALS - LINK_AGORIC_DECIMALS)),
+  )
+  const roundUp = digits[significant.length] && parseInt(digits[significant.length], 10) >= 5
+  let requiredFee = Nat(parseInt(significant, 10))
+  if (roundUp) {
+    requiredFee += 1
+  }
+  return Nat(requiredFee)
+}
+
+const send = (obj) =>
+  new Promise((resolve, reject) => {
+    const data = JSON.stringify(obj)
+    const req = http.request(
+      {
+        hostname: oracleUrl.hostname,
+        port: oracleUrl.port,
+        path: oracleUrl.pathname,
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Content-Length': data.length,
+        },
+      },
+      (res) => {
+        if (res.statusCode === 200) {
+          resolve(res.statusCode)
+        } else {
+          reject(res.statusCode)
+        }
+      },
+    )
+    req.on('error', reject)
+    req.write(data)
+    req.end()
+  })
+
+const execute = async (request) => {
+  const validator = new Validator(request, customParams)
```

Comment:
```suggestion
  const data = JSON.parse(request.data.result)
  request.data = data
  const validator = new Validator(request, customParams)
```
This should get you the data sent from a previous task.

Reply:
Better yet if you put a JsonParse adapter _before_ the Agoric adapter, you don't have to do the parsing in this adapter. You'll only have to grab the values you want from `request.data.result`.

Reply:
Thanks, I got the `request.data.result` correctly already and returned it to the chain.  I still don't know how to affect the display of `'Task Run Data'` for the Agoric task, which always seems to be `''`.  If I reset the `request.data` before calling the validator, won't that break the other data items I need, like `data.request_id` and `data.payment`?

![Screen Shot 2020-10-26 at 9 49 11 PM](https://user-images.githubusercontent.com/457244/97254917-17d15600-17d5-11eb-8cc9-bacc33292a01.png)